flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/batch/connectors/hive/HiveTableOutputFormat.java

```diff
@@ -0,0 +1,419 @@
+	}
+
+	//  Custom serialization methods
+
+	private void writeObject(ObjectOutputStream out) throws IOException {
+		super.write(out);
+		jobConf.write(out);
+		out.writeObject(isPartitioned);
+		out.writeObject(isDynamicPartition);
+		out.writeObject(overwrite);
+		out.writeObject(rowTypeInfo);
+		out.writeObject(hiveTablePartition);
+		out.writeObject(partitionCols);
+		out.writeObject(dbName);
+		out.writeObject(tableName);
+		out.writeObject(tblProperties);
+	}
+
+	@SuppressWarnings("unchecked")
+	private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
+		super.read(in);
+		if (jobConf == null) {
+			jobConf = new JobConf();
+		}
+		jobConf.readFields(in);
+		jobConf.getCredentials().addAll(this.credentials);
+		Credentials currentUserCreds = HadoopInputFormatCommonBase.getCredentialsFromUGI(UserGroupInformation.getCurrentUser());
+		if (currentUserCreds != null) {
+			jobConf.getCredentials().addAll(currentUserCreds);
+		}
+		isPartitioned = (boolean) in.readObject();
+		isDynamicPartition = (boolean) in.readObject();
+		overwrite = (boolean) in.readObject();
+		rowTypeInfo = (RowTypeInfo) in.readObject();
+		hiveTablePartition = (HiveTablePartition) in.readObject();
+		partitionCols = (List<String>) in.readObject();
+		dbName = (String) in.readObject();
+		tableName = (String) in.readObject();
+		partitionToWriter = new HashMap<>();
+		tblProperties = (Properties) in.readObject();
+	}
+
+	@Override
+	public void finalizeGlobal(int parallelism) throws IOException {
+		StorageDescriptor jobSD = hiveTablePartition.getStorageDescriptor();
+		Path stagingDir = new Path(jobSD.getLocation());
+		FileSystem fs = stagingDir.getFileSystem(jobConf);
+		IMetaStoreClient client = null;
+		try {
+			client = HMSClientFactory.create(new HiveConf(jobConf, HiveConf.class));
+			Table table = client.getTable(dbName, tableName);
+			if (!isDynamicPartition) {
+				commitJob(stagingDir.toString());
+			}
+			if (isPartitioned) {
+				// TODO: to be implemented
+			} else {
+				moveFiles(stagingDir, new Path(table.getSd().getLocation()));
+			}
+		} catch (TException e) {
+			throw new CatalogException("Failed to query Hive metaStore", e);
+		} finally {
+			if (client != null) {
+				client.close();
+			}
+			fs.delete(stagingDir, true);
+		}
+	}
+
+	@Override
+	public void initializeGlobal(int parallelism) throws IOException {
+	}
+
+	@Override
+	public void configure(Configuration parameters) {
+		// since our writers are transient, we don't need to do anything here
+	}
+
+	@Override
+	public void open(int taskNumber, int numTasks) throws IOException {
+		try {
+			StorageDescriptor sd = hiveTablePartition.getStorageDescriptor();
+			serializer = (AbstractSerDe) Class.forName(sd.getSerdeInfo().getSerializationLib()).newInstance();
+			ReflectionUtils.setConf(serializer, jobConf);
+			// Pass null as partition properties, assuming they're same as table properties
```

Comment:
I think it's possible that a Hive partition can be different from its table in terms of properties. For instance, a partition can has a different storage format. For now, I think we can add a todo item.